src/Microsoft.AspNet.OData/Adapters/WebApiActionMap.cs

```diff
@@ -41,5 +41,29 @@ public bool Contains(string name)
         {
             return this.innerMap.Contains(name);
         }
+
+        /// <summary>
+        /// Gets the action descriptor of the specified action
+        /// </summary>
+        /// <param name="actionName">The name of the action</param>
+        /// <returns>The <see cref="IWebApiActionDescriptor"/> if it exists, otherwise null</returns>
+        public IWebApiActionDescriptor GetActionDescriptor(string actionName)
+        {
+            var group = this.innerMap.FirstOrDefault(g => g.Key == actionName);
```

Comment:
@habbes Confirm if a scenario where `g.Key` and `actionName` can differ by case can arise here

Reply:
Since actions are named after their controller methods, I don't think we should perform case-insensitive lookup.

Reply:
But I guess this depends on the implementation of the `innerMap.Contains()` method. It implements `ILookup.Contains()`, but from that alone we can't tell whether it's case-sensitive or not. I can try to test it in the debugger to see if it's case-insensitive. But I'd be surprised if it is.